sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SerializableComparatorTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+import java.io.Serializable;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link SerializableComparator}. */
+@RunWith(JUnit4.class)
+public class SerializableComparatorTest {
+
+  /** Basic test of {@link SerializableComparator} being {@link java.io.Serializable}. */
+  @Test
+  public void testIfImplementsSerializable() {
+    SerializableFunction<String, Integer> fn = Integer::parseInt;
+
+    SerializableComparator<String> cmp = SerializableComparator.comparing(fn);
+    Assert.assertTrue(cmp instanceof Serializable);
```

Comment:
This does not imply that the instance can be serialized. It would be better to use the `SerializableUtils` to check the serialization and deserialization.

Reply:
You could have left the original tests as well, those were fine.

Reply:
Would it be sufficient to check if the returned `Comparator` is serializable using `org.apache.beam.sdk.util.SerializableUtils#ensureSerializable(T)` or should we check using something else?

Reply:
Yes, that is exactly the method to use. :+1:

Reply:
Re-added the original tests and used `org.apache.beam.sdk.util.SerializableUtils#ensureSerializable(T)`.